doc/modules/compose.rst

```diff
@@ -527,6 +527,20 @@ above example would be::
                                   ('countvectorizer', CountVectorizer(),
                                    'title')])
 
+If :class:`~sklearn.compose.ColumnTransformer` is fitted with a dataframe
+and the dataframe only has string column names, then transforming a dataframe
+will use the column names to select the columns::
```

Comment:
```suggestion
and the dataframe has only string column names, then transforming a dataframe
will use the column names to select the columns, no matter in which order they
are::
```
The dataframe in transform also needs string column names, right?

Reply:
This PR doesn't check for the dataframe in `transform` to only string columns. The only requirement was for the dataframe in `transform` to have all the (str) column names that were required by `fit`.

I lean slightly toward the current behavior, but I am open to requiring all strings column in `transform` as well.